hpeOneView/resources/servers/id_pools.py

```diff
@@ -23,64 +23,84 @@
 
 standard_library.install_aliases()
 
-from hpeOneView.resources.resource import ResourceClient
+from hpeOneView.resources.resource import Resource, ResourceSchemaMixin
 
 
-class IdPools(object):
+class IdPools(Resource, ResourceSchemaMixin):
     """
     Class for Id Pools API client.
     """
     URI = '/rest/id-pools'
 
-    def __init__(self, con):
-        self._client = ResourceClient(con, self.URI)
+    def __init__(self, connection, data=None):
+        super(IdPools, self).__init__(connection, data)
 
-    def get(self, id_or_uri):
+    def get(self):
         """
         Gets a pool.
-
         Args:
             id_or_uri: Can be either the range ID or URI.
-
         Returns:
             dict: Pool resource.
         """
-        return self._client.get(id_or_uri)
+        return self._helper.do_get(self.URI)
 
-    def enable(self, information, id_or_uri, timeout=-1):
+    def schema(self):
         """
-        Enables or disables a pool.
+        Gets schema of ID pools and returns it
 
         Args:
-            information (dict): Information to update.
-            id_or_uri: ID or URI of range.
-            timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
-                in OneView; it just stops waiting for its completion.
+            uri:
+                URI of range.
+        Returns:
+            dict: A dict containing the schema.
+        """
+        return self._helper.do_get(self.URI + "/schema")
+
+    def get_pool_type(self, uri):
+        """
+        Gets a pool along with the list of ranges present in it
+
+        Args:
+          uri: URI of range
 
         Returns:
-            dict: Updated resource.
+          dict: List of ranges
+        """
+        uri = self._helper.build_uri(uri)
+        return self._helper.do_get(uri)
+
+    def update_pool_type(self, data, uri, timeout=-1):
```

Comment:
Don't expect user to provide entire uri. Instead, you can collect "pool_type" and frame the uri inside this method.
```
def update_pool_type(self, data, pool_type, timeout=-1):
```

Reply:
We are not getting the entire uri from user... here uri is just pool_type. Later i will get appeded with self.URI